### ISSUE TITLE
Migration for 1.5.0 broken

### ISSUE THREAD

**Issue by axelfontaine** [tembo-io/pgmq#362]
In https://github.com/tembo-io/pgmq/blob/main/pgmq-extension/sql/pgmq--1.4.5--1.5.0.sql

- The actual implementation of `send` (https://github.com/tembo-io/pgmq/blob/main/pgmq-extension/sql/pgmq--1.4.5--1.5.0.sql#L610) should come before the simpler overloads at https://github.com/tembo-io/pgmq/blob/main/pgmq-extension/sql/pgmq--1.4.5--1.5.0.sql#L565
- The actual implementation of `send_batch` (https://github.com/tembo-io/pgmq/blob/main/pgmq-extension/sql/pgmq--1.4.5--1.5.0.sql#L678) should come before the simpler overloads at https://github.com/tembo-io/pgmq/blob/main/pgmq-extension/sql/pgmq--1.4.5--1.5.0.sql#L633

WIthout this change applying it fails with

```
SQL State  : 42883
Error Code : 0
Message    : ERROR: function pgmq.send(text, jsonb, unknown, timestamp with time zone) does not exist
  Hint: No function matches the given name and argument types. You might need to add explicit type casts.
```

and

```
SQL State  : 42883
Error Code : 0
Message    : ERROR: function pgmq.send_batch(text, jsonb[], unknown, timestamp with time zone) does not exist
  Hint: No function matches the given name and argument types. You might need to add explicit type casts.
```

respectively.

What really should happen here, is that all the migrations should be automatically integration tested  as part of the release process to prevent this from occurring again.

**Comment by ChuckHend** [comment]
Thanks for raising the issue @axelfontaine . There is a [CI job](https://github.com/tembo-io/pgmq/blob/main/.github/workflows/extension_upgrade.yml) that checks the upgrade path from v1.0.0, but what you've shown tells me there must be a bug in how that test is being executed, or we are just missing test coverage.

**Comment by ChuckHend** [comment]
@axelfontaine do you want to get a PR up to resolve this? We can release v1.5.1 quickly afterwards, then work on fixing how these tests work.

**Comment by ChuckHend** [comment]
@axelfontaine, I opened a [PR](https://github.com/tembo-io/pgmq/pull/363), but I haven't been able to quite reproduce this yet.

You're seeing this error when calling send and send_batch using one of the new 4 parameter variants, correct? If so, could you confirm exactly which values you are using? If you're having an issue with the `delay` parameter needs to be explicitly typed, so either pass in an integer like `10` or explicitly cast it `'10'::integer`.

**Comment by axelfontaine** [comment]
No, I am seeing the error when running to 1.5.0 migration against a 1.4.5 schema.

**Comment by ChuckHend** [comment]
can you provide the output you get when you run it? something like:

```
\dx pgmq
                                 List of installed extensions
 Name | Version | Schema |                             Description                             
------+---------+--------+---------------------------------------------------------------------
 pgmq | 1.4.5   | pgmq   | A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.
(1 row)

alter extension pgmq update to '1.5.0';
ALTER EXTENSION
postgres=# \dx pgmq
                                 List of installed extensions
 Name | Version | Schema |                             Description                             
------+---------+--------+---------------------------------------------------------------------
 pgmq | 1.5.0   | pgmq   | A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.
(1 row)
```

**Comment by ChuckHend** [comment]
@axelfontaine , can you confirm how you are executing the migration script? They are not meant to be executed directly, rather by `alter extension pgmq update`

**Comment by axelfontaine** [comment]
@ChuckHend Sorry for the late reply. I've been busy with the launch of our new product https://sprinters.sh (10x cheaper GitHub Actions on your own AWS account). I'll get back to this in a few days.
